js/files-tabview.js

```diff
@@ -21,99 +22,164 @@
-                    for (var i = 0; i < arrayLength; i++) {
-                        html += this.insertPersonRow(data.persons[i]);
-                    }
-                    html += "</table>";
-                }
-                else
-                {
-                    html += "<div class='emptycontent'>";
-                    html += "<div class='icon-user svg'></div>";
-                    html += "<p>"+t('facerecognition', 'No people found')+"</p>";
-                    html += "</div>";
+            var arrayLength = persons.length;
+            if (arrayLength > 0) {
+                html += "<table class='persons-list'>";
+                for (var i = 0; i < arrayLength; i++) {
+                    html += this.getPersonRowTemplate(persons[i]);
                 }
+                html += "</table>";
             }
-            else
-            {
+            else {
                 html += "<div class='emptycontent'>";
                 html += "<div class='icon-user svg'></div>";
-                html += "<p>"+t('facerecognition', 'This image is not yet analyzed')+"</p>";
-                html += "<p><span>"+t('facerecognition', 'Please, be patient')+"</span></p>";
+                html += "<p>"+t('facerecognition', 'No people found')+"</p>";
                 html += "</div>";
             }
+            return html;
+        },
+
+        getUserDisabledTemplate: function () {
+            var openSettingsLink = t('facerecognition', 'Open <a target="_blank" href="{settingsLink}">settings ↗</a> to enable it.',
+                                    {settingsLink: OC.generateUrl('settings/user/facerecognition')});
+            var html = "";
+            html += "<div class='emptycontent'>";
+            html += "<div class='icon-user svg'></div>";
+            html += "<p>" + t('facerecognition', 'Facial recognition is disabled') + "</p>";
+            html += "<p><span>" + openSettingsLink + "</span></p>";
+            html += "</div>";
+            return html;
+        },
+
+        getFolderDisabledTemplate: function () {
+            var html = "";
+            html += "<div class='emptycontent'>";
+            html += "<div class='icon-user svg'></div>";
+            html += "<p>" + t('facerecognition', 'Facial recognition is disabled for this folder.') + "</p>";
+            html += "</div>";
+            return html;
+        },
+
+        getFolderTemplate: function (data) {
+            var html = "";
+            html += "<div class='emptycontent'>";
+            html += "<div class='icon-user svg'></div>";
+            html += "<p>";
+            html += "<input class='checkbox' id='searchPersonsToggle'";
+            if (data.child_detection)
+                html += "checked='checked'";
+            html += "type='checkbox'>";
+            html += "<label for='searchPersonsToggle'>" + t('facerecognition', 'Search for my loved ones in this folder.') + "</label>";
```

Comment:
this "loved ones" seems very patronizing to me (and can be far out of context). I would keep it neutral ("search for faces"/"apply face recognition"/...)

Reply:
haha.. :sweat_smile: 
I always thought the same .. but I had based on google photos.

Reply:
Maybe .. 'Search for persons in the images of this directory.'